### PR TITLE
Added aks module for k8s

### PR DESCRIPTION
**Pull request by dunefro**

### CODE REVIEW

File: aks.tf [hunk 1]
```diff
@@ -11,29 +11,69 @@ resource "azurerm_role_assignment" "network_contributor_cluster" {
...
+  enable_auto_scaling                        = true
+  enable_host_encryption                     = true
+  identity_ids                               = [azurerm_user_assigned_identity.cluster.id]
+  identity_type                              = "UserAssigned"
+  kubernetes_version                         = var.kubernetes_version
+
+  # network configuration
+  network_plugin             = var.network_plugin
+  vnet_subnet_id             = var.subnet_id
+  net_profile_dns_service_ip = var.dns_ip
```
Comment (shubhamrai1993) [inline aks.tf:58]: Does the DNS IP necessarily have to be taken in as an input? Its optional and I think will be rarely changed

File: aks.tf [hunk 2]
```diff
@@ -11,29 +11,69 @@ resource "azurerm_role_assignment" "network_contributor_cluster" {
...
-  local_account_disabled = false
-  dns_prefix             = var.name
-  identity {
-    type         = "UserAssigned"
-    identity_ids = [azurerm_user_assigned_identity.cluster.id]
-  }
-  network_profile {
-    network_plugin = "kubenet"
+module "aks" {
+  source                    = "Azure/aks/azurerm"
```
Comment (shubhamrai1993) [inline aks.tf:15]: Is there a reason to use this module in place of `azurerm_kubernetes_cluster` directly?

File: locals.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  tags = merge(
+    {
+      "terraform-module" = "network"
+      "terraform"        = "true"
+      "cluster-name"     = var.name
+    },
+    var.tags
+  )
+  intial_node_pool_instance_count = var.control_plane ? 2 : 1
+  node_pools = {
```
Comment (shubhamrai1993) [inline locals.tf:11]: Can we add provision for additional agent pools in the module? Right now we will have to go around terraform if we need an additional agent pool it seems

File: locals.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      name                = "spotpool"
+      node_count          = 1
+      max_count           = 20
+      min_count           = 1
+      os_disk_size_gb     = 100
+      priority            = "Spot"
+      vm_size             = var.intial_node_pool_spot_instance_type
+
+      # mandatory to pass otherwise node pool will be recreated
+      enable_auto_scaling = true
```
Comment (shubhamrai1993) [inline locals.tf]: formatting in this section looks off

File: locals.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
+locals {
+  tags = merge(
+    {
+      "terraform-module" = "network"
```
Comment (shubhamrai1993) [inline locals.tf]:
```suggestion
      "terraform-module" = "terraform-azure-truefoundry-cluster"
```

File: aks.tf (hunk 1 above)
Comment (dunefro) [reply aks.tf:58]: this is kept if the CIDR changes, then DNS IP can also be changed.

File: aks.tf (hunk 2 above)
Comment (dunefro) [reply aks.tf:15]: Better support, versioning, and internal sub-resources gets automatically created.

**Review by shubhamrai1993 — APPROVED** [review]

**Pull request merged**